yew/src/agent/worker/queue.rs

```diff
@@ -0,0 +1,48 @@
+use std::any::TypeId;
+use std::cell::{RefCell, RefMut};
+use std::collections::{hash_map, HashMap, HashSet};
+
+/// Thread-local instance used to queue worker messages
+pub struct Queue {
+    loaded_agents: RefCell<HashSet<TypeId>>,
+    msgs_queue: RefCell<HashMap<TypeId, Vec<Vec<u8>>>>,
+}
+
+impl Queue {
+    pub fn new() -> Queue {
+        Queue {
+            loaded_agents: RefCell::new(HashSet::new()),
+            msgs_queue: RefCell::new(HashMap::new()),
+        }
+    }
+
+    pub fn borrow_msg_queue_mut(&self) -> RefMut<'_, HashMap<TypeId, Vec<Vec<u8>>>> {
```

Comment:
If we keep these (this needs to be read in context of the comment just below this one 😄) I think it would be worth adding `#[inline(always)]` to them (not sure if it actually makes a difference – we might need to experiment and look at the compiler's output).

Reply:
I've opted for `#[inline]` for now, I am partial on the abstraction through functions though, as it allows for the fields that we access to stay private. I do think mayhaps `msg_queue` can just be public, but on the other hand with `#[inline]` that wouldn't be a performance benefit or anything.

So I'll leave the decision on methods vs direct access where needed up to the maintainers

Reply:
Yeah that's true. I see the benefits of using functions. 🤷🏼